packages/amplify-util-mock/src/CFNParser/resource-processors/appsync-resource-processors.ts

```diff
@@ -0,0 +1,257 @@
+    ApiId: apiId,
+    GraphQLUrl: 'http://localhost:20002/',
+    ...(resource.Properties.AdditionalAuthenticationProviders
+      ? {
+          additionalAuthenticationProviders: resource.Properties.AdditionalAuthenticationProviders.map(p => {
+            return {
+              authenticationType: p.AuthenticationType,
+              ...(p.OpenIDConnectConfig ? { openIDConnectConfig: p.OpenIDConnectConfig } : {}),
+              ...(p.CognitoUserPoolConfig ? { cognitoUserPoolConfig: p.CognitoUserPoolConfig } : {}),
+            };
+          }),
+        }
+      : {
+          additionalAuthenticationProviders: [],
+        }),
+  };
+  return processedResource;
+}
+
+export type AppSyncAPIKeyProcessedResource = CloudFormationProcessedResourceResult & {
+  ApiKey: string;
+  Ref: string;
+};
+export function appSyncAPIKeyResourceHandler(
+  resourceName,
+  resource,
+  cfnContext: CloudFormationParseContext,
+  transformResult: any,
+): AppSyncAPIKeyProcessedResource {
+  const value = 'da2-fakeApiId123456'; // TODO: Generate
+  const processedResource = {
+    cfnExposedAttributes: { ApiKey: 'ApiKey', Arn: 'ref' },
+    ApiKey: value,
+    Ref: `arn:aws:appsync:us-east-1:123456789012:apis/graphqlapiid/apikey/apikeya1bzhi${value}`,
+    Arn: `arn:aws:appsync:us-east-1:123456789012:apis/graphqlapiid/apikey/apikeya1bzhi${value}`,
+  };
+  return processedResource;
+}
+
+export type AppSyncSchemaProcessedResource = CloudFormationProcessedResourceResult & {
+  content: string;
+  path: string;
+};
+export function appSyncSchemaHandler(
+  resourceName,
+  resource,
+  cfnContext: CloudFormationParseContext,
+  transformResult: any,
+): AppSyncSchemaProcessedResource {
+  return {
+    cfnExposedAttributes: {},
+    content: transformResult.schema,
+    path: 'schema.json', // XXX: handle schema folder
+  };
+}
+
+export type AppSyncResolverProcessedResource = CloudFormationProcessedResourceResult & {
+  dataSourceName?: string;
+  functions: string[];
+  typeName: string;
+  fieldName: string;
+  requestMappingTemplateLocation: string;
+  responseMappingTemplateLocation: string;
+  ResolverArn: string;
+  kind: 'UNIT' | 'PIPELINE';
+};
+
+export function appSyncResolverHandler(
+  resourceName,
+  resource,
+  cfnContext: CloudFormationParseContext,
+  transformResult: any,
+): AppSyncResolverProcessedResource {
+  const { Properties: properties } = resource;
+  const requestMappingTemplate = parseValue(properties.RequestMappingTemplateS3Location, cfnContext).replace(
+    's3://${S3DeploymentBucket}/${S3DeploymentRootKey}/',
+    '',
+  );
+  const responseMappingTemplate = parseValue(properties.ResponseMappingTemplateS3Location, cfnContext).replace(
+    's3://${S3DeploymentBucket}/${S3DeploymentRootKey}/',
+    '',
+  );
+  let dataSourceName;
+  let functions;
+  if (properties.Kind === 'PIPELINE') {
+    functions = (properties.PipelineConfig.Functions || []).map(f => parseValue(f, cfnContext));
```

Comment:
Is anything validates if properties has PipelineConfig before this is getting called? or no need to validate because higher level a template like that could not be built?